### ISSUE TITLE
Doesn match specifier

### ISSUE THREAD

**Issue by apoellitsi1926** [allinurl/goaccess#2052]
Hi all,

I am getting the following error:
21-03-01 00' doesn't match specifier '%d'

Previously i was getting the error that my specified IP address doesn't match specifier '%h' but i think i managed to resolve this. Now i am getting this one and i tried different ways but still the same.

This is how it looks like:

`lo-ohs2.services.test,xxx.xx.xxx.xxx - www.test.org.de npeter3 [2021-03-01 00:00:00.734099 +0200] 005j6mVR8jgBT8h5xBo2yW0005HO003DXI "GET /api/tasks/count?dateFrom=2020-03-01&dateTo=2021-03-01 HTTP/1.1" 200 37 1830 527 16794 17010 "https://test.test.org.de/dashboard/care-me/test" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36"

lo-ohs1.services.test,xxx.xxx.xxx.xx - www.test.org.de - [2021-03-01 00:00:00.885744 +0200] 005j6mVRhk^BT8h5xBp2iW0000vy0000OY "GET /assets/i18n/el.json HTTP/1.1" 200 1591 578 1982 287 872 "https://www.test.org.de/launchpad.html" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36"
`

**Comment by 0bi-w6n-K3nobi** [comment]
Hi @apoellitsi1926.

You must declare parts that must be ignored in your `LOG`.
In this `LOG`, my initial bet would be:

`--log-format='%^,%h ...'`

I'm so confuse about your fields declaration like `%{begin`, `%{Refer`, etc.
I believe that it is wrong.
I ask you to wait a little and I will help you. 

You can see more details in manual [here](https://goaccess.io/man) at `SPECIFIERS`.

**Comment by apoellitsi1926** [comment]
Hi,
Thanks for your reply.

The specific entries are as follows, in Oracle HTTP Server / Apache notation:

    %h – Remote client IP
    %l – Remote logname (always dash)
    %v – Virtual host - this is the base application URL
    %u – Remote user, if the request is authenticated
    [%{begin:%F %T}t.%{begin:usec_frac}t %{%z}t] – Timestamp of beginning of client request including microseconds fraction + timezone.  Timezone is local.
    %E – Oracle ECID (execution context id, this is a  GUID assigned by the Oracle HTTP server to trace the request across execution tiers)
    "%r – First line of request (typically HTTP verb + URL)
    %>s – HTTP status code
    %b – Size of response in bytes excluding HTTP response headers
    %I – Bytes received, includes headers + request
    %O – Bytes sent, includes headers + reply
    %^FB - Delay in microseconds between the beginning of request arrival and beginning to write back the reply (includes request arrival time)
    %D – Time taken to service the request in microseconds - includes request arrival and full response transmit time
    \"%{Referer}i\"  - Referer URL
    \"%{User-Agent}i\"" – Browser Agent signature

**Comment by 0bi-w6n-K3nobi** [comment]
Hi @allinurl .

I made a small mistake here induced by the format made by the user ... so I used `... %u ... %u ...`.
Ok ... The version **1.4.3** correctly accused the error:
`Token already set for '% u' specifier.`

I checked for versions **1.4.4** and **14.6** ... It did **NOT** present this error.

**Comment by 0bi-w6n-K3nobi** [comment]
Hi @apoellitsi1926 .

My bet for format is:
`--log-format='%^,%h %^ %v %e [%d %t.%^] %^ "%r" %s %b %^ %^ %^ %D "%R" "%u"'`

**notes:**

In `GOACCES` there are no fields corresponding to your fields `%E`, `%I`, `%O` and `%^FB`.

**Comment by apoellitsi1926** [comment]
Hi i am still getting the same error. 

As the fields %E, %I, %O and %^FB are not supported by GOACCESS then it means that  can't use it to visualize my logs :(

**Comment by 0bi-w6n-K3nobi** [comment]
Hi @apoellitsi1926 .

Did you remove `--log-format=COMBINED`?
Because it will be redefine LogFormat to `COMBINED` format and will remove your customization.

Otherwise... What version do you running?

**Comment by 0bi-w6n-K3nobi** [comment]
Hi @apoellitsi1926 .

Like a workaround for fields `%E`, `%I`, `%O` and %^FB -- You can running a custom LogFormat for each, and visualize into different reports.
For field:
* `%E` -- you can use referrer field `%R` -- well, this field has a lot of variation in value - Are you sure you want to view it?
* `%I` and `%O` -- use field `%b`;
* %^FB -- use field `%D`;

Ok... It is a simple suggestion.

**Comment by 0bi-w6n-K3nobi** [comment]
Hi @allinurl .

Did you see my note/issue below?
You do not need to answer. Just pick your icon reaction.

**Comment by allinurl** [comment]
@0bi-w6n-K3nobi I agree, right now for custom fields that are not currently supported, using any of the "free" fields should do it as a workaround. There's an open issue for custom field, the plan is to support fields such as `%1`, `%2`, etc.

**Comment by apoellitsi1926** [comment]
Hi @0bi-w6n-K3nobi ,

You were right about the --log-format=COMBINED but i still get the following error:
Token '' doesn't match specifier '%h'

I am using this:
--log-format='%^,%h %^ %v %e [%d %t.%^] %^ "%r" %s %b %^ %^ %^ %D "%R" "%u"'

**Comment by allinurl** [comment]
@apoellitsi1926 The format posted by @0bi-w6n-K3nobi works fine for those two lines you posted above. Can you attach a good chunk of your access log so we can make sure you got the right format?

![2021-03-17-081917_841x415_scrot](https://user-images.githubusercontent.com/5005367/111473950-8a49c080-86f9-11eb-998a-013168f7a268.png)

**Comment by apoellitsi1926** [comment]
Hi @allinurl ,

The last line of my log was corrupted that is why i was getting an error.
It runs perfectly now.

Thank you both for your support @0bi-w6n-K3nobi

**Comment by 0bi-w6n-K3nobi** [comment]
Hi @allinurl .

And about [this](https://github.com/allinurl/goaccess/issues/2052#issuecomment-796888047) issue? About the latest versions no showing errors.

**Comment by allinurl** [comment]
@0bi-w6n-K3nobi Sorry I didn't see that post but I see the issue now. Like you said, before it used to display that error, I believe I removed this thinking that setting it twice would simply take the first one. This was done after JSON support was added. Let me take a look if I can bring that back. Thanks again for pointing this out.
